app/models/service_load_balancer.rb

```diff
@@ -0,0 +1,68 @@
+class ServiceLoadBalancer < Service
+  include ServiceLoadBalancerMixin
+  include ServiceLoadBalancerOptionsMixin
+
+  # read from DB or parse from dialog
+  def load_balancer_name
+    @load_balancer_name ||= get_option(:load_balancer_name) || OptionConverter.load_balancer_name(get_option(:dialog) || {})
+  end
+
+  # override existing load_balancer name (most likely from dialog)
+  def load_balancer_name=(lbname)
+    @load_balancer_name = lbname
+    save_option(:load_balancer_name, lbname)
+  end
+
+  def load_balancer_status
+    return "check_status_failed", "load_balancer has not been deployed" unless load_balancer
+
+    'create_complete'
+  rescue MiqException::MiqLoadBalancerNotExistError, MiqException::MiqLoadBalancerStatusError => err
+    # naming convention requires status to end with "failed"
+    ["check_status_failed", err.message]
+  end
+
+  def deploy_load_balancer
```

Comment:
@Ladas I have refactored the `OrchestrationService` class in #11203. Please follow the patter there. `add_resource` should be set in a `post_provision_configure` step.